src/bundle/ui-dev/src/modules/universal-discovery/components/breadcrumbs/breadcrumbs.js

```diff
@@ -88,14 +93,24 @@ const Breadcrumbs = () => {
     };
 
     useEffect(() => {
-        if (hiddenListVisible) {
-            window.document.body.addEventListener('click', toggleHiddenListVisible, false);
-        } else {
-            window.document.body.removeEventListener('click', toggleHiddenListVisible, false);
+        if (!hiddenListVisible) {
+            return;
         }
 
-        return () => window.document.body.removeEventListener('click', toggleHiddenListVisible, false);
-    }, [hiddenListVisible, toggleHiddenListVisible]);
+        const hideHiddenMenuOnClickOutside = (event) => {
+            const { target } = event;
+
+            if (hiddenListWrapperRef.current?.contains(target) ?? false) {
```

Comment:
in theory `?? false` is unnecessary, I think? as if it does not exist, it's undefined, which is falsy as well

Reply:
I wanted to be explicit here so that it will be easier to understand when reading.